db/db_impl/db_impl_write.cc

```diff
@@ -1959,7 +1975,70 @@ void DBImpl::NotifyOnMemTableSealed(ColumnFamilyData* /*cfd*/,
 // REQUIRES: this thread is currently at the front of the 2nd writer queue if
 // two_write_queues_ is true (This is to simplify the reasoning.)
 Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context,
-                              std::string replication_sequence) {
+                              const MemTableSwitchRecord& mem_switch_record) {
```

Comment:
There are two many if/else checks if we change the `SwitchMemtable` directly. So I instead implemented another function, which is much simpler since we don't need to care about creating wal, error recovery when creating wal fails, track_and_verify_wals_in_manifest, etc.